dist/tools/buildsystem_sanity_check/check.sh

```diff
@@ -334,6 +334,32 @@ check_no_pkg_source_local() {
         | error_with_message "Don't push PKG_SOURCE_LOCAL definitions upstream"
 }
 
+check_shell_which() {
+    local patterns=()
+    local pathspec=()
+
+    patterns+=(-e '(shell[[:blank:]]\+which')
+
+    pathspec+=('Makefile*')
+    pathspec+=('**/Makefile*')
+    pathspec+=('**/*.mk')
+    git -C "${RIOTBASE}" grep -n "${patterns[@]}" -- "${pathspec[@]}" \
+        | error_with_message "Don't use \`which\` in makefiles, use \`command -v\` instead."
+}
+
+check_stderr_null() {
+    local patterns=()
+    local pathspec=()
+
+    patterns+=(-e '2>[[:blank:]]*&1[[:blank:]]*>[[:blank:]]*/dev/null')
+
+    pathspec+=('Makefile*')
+    pathspec+=('**/Makefile*')
+    pathspec+=('**/*.mk')
+    git -C "${RIOTBASE}" grep -n "${patterns[@]}" -- "${pathspec[@]}" \
+        | error_with_message "Redirecting stderr and stdout to /dev/null is \`>/dev/null 2>&1\`; the other way round puts the old stderr to the new stdout."
```

Comment:
I am not sure this can be generally assumed. I myself, actually, sometimes use `2>&1 > /dev/null` where one would use `&> /dev/null` in `bash` (which is a bashism that does not work in every shell, such as `dash`)

Reply:
`2>&1 >/dev/null` sends the old stderr to the original stdout.

`>/dev/null 2>&1` does as `&>/dev/null` and sends both to /dev/null.

This is what caused the errors in #16776 when output to stderr was suddenly misdirected into a stdout where it was to be compared to `echo $?`'s "0" output.

Reply:
Ah, gotcha!